libcloud/compute/drivers/openstack.py

```diff
@@ -335,6 +335,34 @@ def ex_hard_reboot_node(self, node):
         """
         return self._reboot_node(node, reboot_type='HARD')
 
+    def ex_list_subnet(self, network_id=None):
+        """
+        List all subnet resources
+        :param        network_id:   filter for subnets for this network ID
+        :type         network_id:   ``str``
+        :rtype:       ``list`` of :class:``OpenStackSubnet``
+        """
+        uri = '/subnets'
+        resp = self.connection.request(uri, method='GET')
+        if resp.status == httplib.NOT_FOUND:
```

Comment:
I assume this means that this functionality is not available?

In any case, it's probably better to return an empty list.